src/main/java/com/example/petbuddybackend/controller/CaretakerController.java

```diff
@@ -39,6 +41,19 @@ public Page<CaretakerDTO> getCaretakers(
         return caretakerService.getCaretakers(pageable, filters);
     }
 
+    @PostMapping("/add-or-edit")
+    @Operation(
+            summary = "Add or edit caretaker profile",
+            description = "Add caretaker profile if it does not exists, also can edit caretaker if it exists by changing caretaker data."
+    )
+    @PreAuthorize("isAuthenticated()")
+    public CaretakerDTO addOrEditCaretaker(
+            @RequestBody @Valid CreateCaretakerDTO caretakerDTO,
+            Principal principal
+    ) {
+        return caretakerService.addOrEditCaretaker(caretakerDTO, principal.getName());
+    }
+
```

Comment:
A w sumie czemu by tego nie rozdzielić na create caretaker i update caretaker?

Reply:
Nwm czy jest sens bo podajesz to samo

Reply:
W sumie

Reply:
bo pomyslalem sobie ze front moze nie miec wszystkich informacji o opiekunie, wiec wysle tylko te co chce zedytowac